ledger/ledger.go

```diff
@@ -638,10 +638,55 @@ func (l *Ledger) LookupAgreement(rnd basics.Round, addr basics.Address) (basics.
 	defer l.trackerMu.RUnlock()
 
 	// Intentionally apply (pending) rewards up to rnd.
-	data, err := l.acctsOnline.LookupOnlineAccountData(rnd, addr)
+	data, err := l.acctsOnline.lookupOnlineAccountData(rnd, addr)
 	return data, err
 }
 
+// GetKnockOfflineCandidates retrieves a list of online accounts who will be
+// checked to a recent proposal or heartbeat. Large accounts are the ones worth checking.
+func (l *Ledger) GetKnockOfflineCandidates(rnd basics.Round, proto config.ConsensusParams) (map[basics.Address]basics.OnlineAccountData, error) {
+	l.trackerMu.RLock()
+	defer l.trackerMu.RUnlock()
+
+	// get state proof worker's most recent list for top N addresses
+	if proto.StateProofInterval == 0 {
+		return nil, nil
+	}
+
+	var addrs []basics.Address
+
+	// special handling for rounds 0-240: return participating genesis accounts
```

Comment:
We could, but many of our tests are in these early rounds, so we wanted things to be as uniform as possible, so @cce was kind enough to write this special case.

Now, it isn't really possible to knock anyone off in the first 320 rounds anyway, since a `keyreg` is required to become `IncentiveEligible` (and thus subject to suspension) and a `keyreg` gives you a "free" 320 round heartbeat.

In the future, we may allow suspensions of non-IncentiveEligible accounts, and perhaps `keyreg` should not give you that heartbeat if you were already online.  So I guess I'd like to keep the special handling.